packages/expo-system-ui/android/src/main/java/expo/modules/systemui/SystemUIModule.kt

```diff
@@ -0,0 +1,262 @@
+    activity.runOnUiThread {
+      val color = colorToHex(activity.window.navigationBarColor)
+      promise.resolve(color)
+    }
+  }
+
+  @ExpoMethod
+  fun setNavigationBarBackgroundColor(color: Int, promise: Promise) {
+    activity.runOnUiThread {
+      activity.window.navigationBarColor = color
+      promise.resolve(null)
+    }
+  }
+
+  @ExpoMethod
+  fun getNavigationBarDividerColor(promise: Promise) {
+    activity.runOnUiThread {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+        val color = colorToHex(activity.window.navigationBarDividerColor)
+        promise.resolve(color)
+      } else {
+        promise.reject("unavailable", "'getNavigationBarDividerColor' is only available on Android API 28 or higher!")
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setNavigationBarDividerColor(color: Int, promise: Promise) {
+    activity.runOnUiThread {
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+        activity.window.navigationBarDividerColor = color
+        promise.resolve(null)
+      } else {
+        promise.reject("unavailable", "'setNavigationBarDividerColor' is only available on Android API 28 or higher!")
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun getStatusBarBackgroundColor(promise: Promise) {
+    activity.runOnUiThread {
+      val color = colorToHex(activity.window.statusBarColor)
+      promise.resolve(color)
+    }
+  }
+
+  @ExpoMethod
+  fun setStatusBarBackgroundColor(color: Int, promise: Promise) {
+    activity.runOnUiThread {
+      activity.window.statusBarColor = color
+      promise.resolve(null)
+    }
+  }
+
+  @ExpoMethod
+  fun getStatusBarForegroundStyle(promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        val style = if (controller.isAppearanceLightStatusBars) "light" else "dark"
+        promise.resolve(style)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setStatusBarForegroundStyle(style: String, promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        when (style) {
+          "light" -> controller.isAppearanceLightStatusBars = false
+          "dark" -> controller.isAppearanceLightStatusBars = true
+          else -> {
+            promise.reject("invalid-value", "Value \"$style\" is not a valid Status Bar foreground style!")
+            return@let
+          }
+        }
+        promise.resolve(null)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun getNavigationBarForegroundStyle(promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        val style = if (controller.isAppearanceLightNavigationBars) "light" else "dark"
+        promise.resolve(style)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setNavigationBarForegroundStyle(style: String, promise: Promise) {
```

Comment:
Maybe the term "appearance" is better to use for something related to the device appearance, otherwise users might confuse this with an API that has proper styling.

Reply:
The term "style" is used a lot for these kind of things, e.g. `blurStyle` for BlurView. 🤷